common/processors/concatenate_all_pipelines.py

```diff
@@ -0,0 +1,110 @@
+"""This processor concatenates all datasets into one."""
+
+import yaml
+
+from logging import warning, info
+from datapackage_pipelines.wrapper import ingest
+from petl import fromdicts, look
+from tabulator import Stream
+from zipfile import BadZipFile, ZipFile
+from datapackage_pipelines.wrapper import spew
+
+from common.utilities import format_to_json
+from common.bootstrap import collect_sources
+from common.config import (
+    FISCAL_METADATA_FILE,
+    FISCAL_SCHEMA_FILE,
+    FISCAL_MODEL_FILE,
+    DATAPACKAGE_FILE,
+    SAMPLE_SIZE,
+    DATA_DIR
+)
+
+
+STREAM_OPTIONS = {
+    'scheme': 'text',
+    'format': 'csv',
+    'encoding': 'utf-8',
+    'headers': 1,
+    'sample_size': SAMPLE_SIZE
+}
+
+
+def format_data_sample(stream):
+    """Return a table representation of a sample of the streamed data."""
+
+    keyed_rows = []
+    for row in stream.sample:
+        keyed_rows.append(dict(zip(stream.headers, row)))
+
+    petl_table = fromdicts(keyed_rows)
+    return repr(look(petl_table, limit=None))
+
+
+def collect_local_datasets(select=None):
+    """Collect all local fiscal datasets."""
+
+    for source in collect_sources(select=select):
+        if source.fiscal_zip_file:
+            info('Found %s', source.fiscal_zip_file.lstrip(DATA_DIR))
+
+            try:
+                with ZipFile(source.fiscal_zip_file) as zipped_files:
+                    filenames = zipped_files.namelist()
+                    filenames.remove(DATAPACKAGE_FILE)
+
+                    for i, filename in enumerate(filenames):
+                        csv_bytes = zipped_files.read(filename)
+                        csv_text = csv_bytes.decode()
+
+                        yield filename, csv_text
+
+                        message = 'Collected resource %s of %s: %s'
+                        info(message, i, source.id, filename)
+
+            except BadZipFile:
+                warning('%s has a bad zip file', source.id)
+
+
+def concatenate(csv_files):
+    """Return a single resource generator for all datasets."""
+
+    nb_files = 0
+
+    for filename, csv_text in csv_files:
+        nb_files += 1
+
+        with Stream(csv_text, **STREAM_OPTIONS) as stream:
+            for row in stream.iter(keyed=True):
+                yield row
```

Comment:
Didn't this concatenated file was supposed to contain only the 5 common columns?